### PR TITLE
[feat] Expose the chunk config of consumer to the reader

### LINKED ISSUE #935: [feature request] Expose the chunk config of consumer to the reader
## Motivation

About the chunk config of the `Consumer`, we need to make it exposed in `Reader`.

https://github.com/apache/pulsar-client-go/blob/23777362f3d132f98eff56c4f511f761227d8cf4/pulsar/consumer_partition.go#L112-L114

### PR DESCRIPTION
**Pull request by CrazyCollin**
Fixes #935 

### Motivation

Support expose the chunk config of consumer to the reader

### Modifications

Add chunk config related filed in `ReaderOptions` struct.

### Verifying this change

This change added tests and can be verified as follows:

- Added test for specified and default chunk options in reader.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)

### CODE REVIEW

**Comment by CrazyCollin** [comment]
Hi, @RobertIndie @shibd ,can you review it for me when you are free, many thanks.

**Review by RobertIndie — COMMENTED** [review]
Thanks for your PR. Please add some tests to verify this PR. 

You can add a test to verify that the configuration of the reader can be fed into the internal consumer `reader.pc`.

File: pulsar/reader_test.go [hunk 1]
```diff
@@ -70,6 +70,49 @@ func TestReaderConfigSubscribeName(t *testing.T) {
...
+		Topic:                       "my-topic1",
+		StartMessageID:              EarliestMessageID(),
+		MaxPendingChunkedMessage:    50,
+		ExpireTimeOfIncompleteChunk: 30 * time.Second,
+		AutoAckIncompleteChunk:      true,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer consumer1.Close()
```
Comment (RobertIndie) [inline pulsar/reader_test.go:92]:
```suggestion
	r1, err := client.CreateReader(ReaderOptions{
		Topic:                       "my-topic1",
		StartMessageID:              EarliestMessageID(),
		MaxPendingChunkedMessage:    50,
		ExpireTimeOfIncompleteChunk: 30 * time.Second,
		AutoAckIncompleteChunk:      true,
	})
	if err != nil {
		t.Fatal(err)
	}
	defer r1.Close()
```
Same to the `consumer2` below.

File: pulsar/reader_test.go [hunk 2]
```diff
@@ -70,6 +70,49 @@ func TestReaderConfigSubscribeName(t *testing.T) {
...
+	consumer1, err := client.CreateReader(ReaderOptions{
+		Topic:                       "my-topic1",
+		StartMessageID:              EarliestMessageID(),
+		MaxPendingChunkedMessage:    50,
+		ExpireTimeOfIncompleteChunk: 30 * time.Second,
+		AutoAckIncompleteChunk:      true,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
```
Comment (RobertIndie) [inline pulsar/reader_test.go:91]: Please use `assert.NotNil(t, err)`

Comment (CrazyCollin) [reply pulsar/reader_test.go:91]: Thank you for your advice, but a little confused to me, shouldn't `assert.Nil` be used here?

File: pulsar/reader.go [hunk 3]
```diff
@@ -92,6 +92,16 @@ type ReaderOptions struct {
...
+
+	// MaxPendingChunkedMessage sets the maximum pending chunked messages. (default: 100)
+	MaxPendingChunkedMessage int
+
+	// ExpireTimeOfIncompleteChunk sets the expiry time of discarding incomplete chunked message. (default: 60 seconds)
+	ExpireTimeOfIncompleteChunk time.Duration
+
+	// AutoAckIncompleteChunk sets whether consumer auto acknowledges incomplete chunked message when it should
+	// be removed (e.g.the chunked message pending queue is full). (default: false)
+	AutoAckIncompleteChunk bool
```
Comment (RobertIndie) [inline pulsar/reader.go:104]:
```suggestion
	// AutoAckIncompleteChunk sets whether reader auto acknowledges incomplete chunked message when it should
	// be removed (e.g.the chunked message pending queue is full). (default: false)
	AutoAckIncompleteChunk bool
```

**Review by shibd — APPROVED** [review]

**Pull request merged**